### PR TITLE
add missing raise for formatting functions 

### LINKED ISSUE #1340: Failing to format file should return an exit code different than 0
Can't use online tool in this case cause this is not reproducible there.

Using v4.4.0-alpha-007 I found a case in which fantomas was not able to format the file (will file that source-code problem separately), throwing this:

```
The following exception occurs while formatting stdin: Fantomas.FormatConfig+FormatException: Formatted content is not valid F# code
   at Fantomas.Extras.FakeHelpers.formatContentInternalAsync@95-5.Invoke(Boolean _arg2) in C:\Users\nojaf\Projects\fantomas\src\Fantomas.Extras\FakeHelpers.fs:line 96
   at Microsoft.FSharp.Control.AsyncPrimitives.CallThenInvokeNoHijackCheck[a,b](AsyncActivation`1 ctxt, FSharpFunc`2 userCode, b result1) in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 404
   at Fantomas.CodeFormatterImpl.isValidFSharpCode@378-7.Invoke(AsyncActivation`1 ctxt) in C:\Users\nojaf\Projects\fantomas\src\Fantomas\CodeFormatterImpl.fs:line 378
   at Microsoft.FSharp.Control.Trampoline.Execute(FSharpFunc`2 firstAction) in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 104
--- End of stack trace from previous location ---
   at Microsoft.FSharp.Control.AsyncResult`1.Commit() in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 337
   at Microsoft.FSharp.Control.AsyncPrimitives.RunSynchronouslyInCurrentThread[a](CancellationToken cancellationToken, FSharpAsync`1 computation) in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 870
   at Microsoft.FSharp.Control.AsyncPrimitives.RunSynchronously[T](CancellationToken cancellationToken, FSharpAsync`1 computation, FSharpOption`1 timeout) in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 878
   at Microsoft.FSharp.Control.FSharpAsync.RunSynchronously[T](FSharpAsync`1 computation, FSharpOption`1 timeout, FSharpOption`1 cancellationToken) in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 1142
   at Program.processSourceString(Boolean isFsiFile, String s, FSharpChoice`2 tw, FormatConfig config) in C:\Users\nojaf\Projects\fantomas\src\Fantomas.CoreGlobalTool\Program.fs:line 113
   at Program.stringToFile@310(Boolean force, Boolean profile, String s, String outFile, FormatConfig config) in C:\Users\nojaf\Projects\fantomas\src\Fantomas.CoreGlobalTool\Program.fs:line 322
Processing ./NodeEffect.Server/Handlers/NodeEffectHandler.fs
./NodeEffect.Server/Handlers/NodeEffectHandler.fs has been written.
Processing ./NodeEffect.Server/Handlers/WatcherHandler.fs
The following exception occurs while formatting stdin: Fantomas.FormatConfig+FormatException: Formatted content is not valid F# code
   at Fantomas.Extras.FakeHelpers.formatContentInternalAsync@95-5.Invoke(Boolean _arg2) in C:\Users\nojaf\Projects\fantomas\src\Fantomas.Extras\FakeHelpers.fs:line 96
   at Microsoft.FSharp.Control.AsyncPrimitives.CallThenInvokeNoHijackCheck[a,b](AsyncActivation`1 ctxt, FSharpFunc`2 userCode, b result1) in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 404
   at Fantomas.CodeFormatterImpl.isValidFSharpCode@378-7.Invoke(AsyncActivation`1 ctxt) in C:\Users\nojaf\Projects\fantomas\src\Fantomas\CodeFormatterImpl.fs:line 378
   at Microsoft.FSharp.Control.Trampoline.Execute(FSharpFunc`2 firstAction) in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 104
--- End of stack trace from previous location ---
   at Microsoft.FSharp.Control.AsyncResult`1.Commit() in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 337
   at Microsoft.FSharp.Control.AsyncPrimitives.RunSynchronouslyInCurrentThread[a](CancellationToken cancellationToken, FSharpAsync`1 computation) in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 870
   at Microsoft.FSharp.Control.AsyncPrimitives.RunSynchronously[T](CancellationToken cancellationToken, FSharpAsync`1 computation, FSharpOption`1 timeout) in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 878
   at Microsoft.FSharp.Control.FSharpAsync.RunSynchronously[T](FSharpAsync`1 computation, FSharpOption`1 timeout, FSharpOption`1 cancellationToken) in F:\workspace\_work\1\s\src\fsharp\FSharp.Core\async.fs:line 1142
   at Program.processSourceString(Boolean isFsiFile, String s, FSharpChoice`2 tw, FormatConfig config) in C:\Users\nojaf\Projects\fantomas\src\Fantomas.CoreGlobalTool\Program.fs:line 113
   at Program.stringToFile@310(Boolean force, Boolean profile, String s, String outFile, FormatConfig config) in C:\Users\nojaf\Projects\fantomas\src\Fantomas.CoreGlobalTool\Program.fs:line 322
```

However, the exit code was still zero, which can then not be picked up by CI pipelines, for example.

### PR DESCRIPTION
**Pull request by btzo**
fix #1340 

Running `Fantomas.CoreGlobalTool` to format a file without the `--check` argument, it does the treatment in the else listed below, calling the appropriate function given type of `inputPath` and `outputPath`.

https://github.com/fsprojects/fantomas/blob/d22dc764910040fd70412c3f3c674c6f4a5a1250/src/Fantomas.CoreGlobalTool/Program.fs#L386-L406

The problem is that when an error occurred in one of these functions, it was treated locally but not propagated, 
so the `try/with` condition that changes the error code to 1 is never achieved.

The ideal solution would be to show only the error messages in the last `try\with` and add tests for each function in the `match` expression, besides the simple test that I inserted.

What do you think @nojaf ?

### CODE REVIEW

File: src/Fantomas.CoreGlobalTool.Tests/CheckTests.fs [hunk 1]
```diff
@@ -28,11 +28,17 @@ let ``formatted files should report exit code 0`` () =
     exitCode |> should equal 0
 
 [<Test>]
-let ``invalid files should report exit code 1`` () =
+let ``invalid files should report exit code 1 with check`` () =
```
Comment (nojaf) [inline src/Fantomas.CoreGlobalTool.Tests/CheckTests.fs:31]: I would remove this `with check` part, it is implied by the file itself.

File: src/Fantomas.CoreGlobalTool.Tests/CheckTests.fs [hunk 2]
```diff
@@ -28,11 +28,17 @@ let ``formatted files should report exit code 0`` () =
...
     use fileFixture = new TemporaryFileCodeSample(WithErrors)
     let (exitCode, _) = checkCode fileFixture.Filename
     exitCode |> should equal 1
 
+[<Test>]
```
Comment (nojaf) [inline src/Fantomas.CoreGlobalTool.Tests/CheckTests.fs:36]: I would move this to a new file `ExitCodeTests` as this is not really related to `--check`.

**Comment by nojaf** [comment]
Hey Mike, that sounds about right.
Thanks for this effort!

File: src/Fantomas.CoreGlobalTool.Tests/CheckTests.fs (hunk 2 above)
Comment (btzo) [reply src/Fantomas.CoreGlobalTool.Tests/CheckTests.fs:36]: Yeah, I think it's better this way.

In that case you think I should replicate the other tests from `CheckTests` to `ExitCodeTests`, replacing the function `checkCode` by the function `formatCode` ?

File: src/Fantomas.CoreGlobalTool.Tests/CheckTests.fs (hunk 1 above)
Comment (btzo) [reply src/Fantomas.CoreGlobalTool.Tests/CheckTests.fs:31]: Fine

File: src/Fantomas.CoreGlobalTool.Tests/CheckTests.fs (hunk 2 above)
Comment (nojaf) [reply src/Fantomas.CoreGlobalTool.Tests/CheckTests.fs:36]: Hmm, I won't force you to replicate all tests. I feel like the one you have covered the need here.
The cli application isn't perfect so in this case, I would blink an eye for not having tests that cover all changes.
Maybe just clean up what you have and we can merge this.

Comment (btzo) [reply src/Fantomas.CoreGlobalTool.Tests/CheckTests.fs:36]: Ok

**Review by nojaf — APPROVED** [review]

**Comment by nojaf** [comment]
Thanks again @btzo!
